src/app/inventory/item-types.ts

```diff
@@ -369,6 +369,18 @@ export interface DimPlug {
   cannotCurrentlyRoll?: boolean;
 }
 
+export interface DimPlugSet {
+  /** The hash that links to a DestinyPlugSetDefinition. */
+  hash: number;
+  /**
+   * A list of built DimPlugs that are found in the plugSet.
+   * This plugs included encompass everything that can be plugged into the socket whether it
+   * is available to the character or not. You should filter this list down based on the plugs
+   * available to the profile/character.
```

Comment:
Should we precalculate a mapping from character to unlock status?

Reply:
We definitely could, where would be the best place to do that?